lib/urlwatch/command.py

```diff
@@ -142,6 +142,21 @@ def test_filter(self, id):
         # (ignore_cached) and we do not want to store the newly-retrieved data yet (filter testing)
         return 0
 
+    def run_new_jobs(self):
+        new_jobs = list()
+        for idx, job in enumerate(self.urlwatcher.jobs):
+            history_data = self.urlwatcher.cache_storage.get_history_data(job.get_guid(), 1)
```

Comment:
Since you only care about whether there's history data or not, a `count_history_data()` or even a `has_history_data()` might speed things up as you mentioned.

Reply:
Maybe, i did not try it since i haven't looked into minidb. But storage.py:588 is not so long that i think there is much to get. But i can try a bit.

Reply:
I have played a but with it and with the current implementation a look up for 1457 Jobs took around 3 Minutes and 24 Seconds. 

So i did try to tone the get_history_data a bit down but it did not has a huge effect. My next attempt was to broaden the Database queue to not just request the information for a specific guid but everything with tries==0 or None. 
If i do that for each job, the time would be 5 time as much so i did try to queue just once and store the guid's in a list and just reuse it. 
That works great in my first tests and reduced the time to process all these jobs to ~0.875620 seconds.

```
    def has_history_data(self, guid):
        if not self._cached_has_history_data_list:
            for data in list(CacheEntry.query(self.db, CacheEntry.c.guid, where=((CacheEntry.c.tries == 0) | (CacheEntry.c.tries == None)))):  # noqa:E711
                self._cached_has_history_data_list.append(data[0])
        try:
            if self._cached_has_history_data_list.index(guid):
                return True
        except ValueError:
            return False
        return False
```

My main issue now is that i never does anything with redis so far. so a redis version needs a bit more time. 

PS: My current Database has 67432 Rows and its size is 1.026.822.144 Bytes